lib/elixir/lib/exception.ex

```diff
@@ -943,30 +941,46 @@ defmodule MismatchedDelimiterError do
     lines = String.split(snippet, "\n")
     expected_delimiter = :elixir_tokenizer.terminator(opening_delimiter)
 
+    start_message = "└ unclosed delimiter"
+    end_message = ~s/└ mismatched closing delimiter (expected "#{expected_delimiter}")/
+
     snippet =
-      format_snippet(
+      SnippetFormatter.format_snippet(
         start_pos,
         end_pos,
         line_offset,
         description,
         file,
         lines,
-        expected_delimiter
+        start_message,
+        end_message
       )
 
     format_message(file, end_line, end_column, snippet)
   end
 
-  defp format_snippet(
-         {start_line, _start_column} = start_pos,
-         {end_line, end_column} = end_pos,
-         line_offset,
-         description,
-         file,
-         lines,
-         expected_delimiter
-       )
-       when start_line < end_line do
+  defp format_message(file, line, column, message) do
+    location = Exception.format_file_line_column(Path.relative_to_cwd(file), line, column)
+    "mismatched delimiter found on " <> location <> "\n" <> message
+  end
+end
+
+defmodule SnippetFormatter do
```

Comment:
Instead of creating a new module, you can put it inside `Exception` with `@doc false` :)